### PR TITLE
Implements mkdirSync

### LINKED ISSUE #593: Implement deno.mkdir()
```
// mkdir creates a new directory with the specified name and permission bits  (before umask).
function mkdir(path: string, mode = 0o777): Promise<void>;
```

With tests - use #586 and #585

### PR DESCRIPTION
**Pull request by sijad**
it's related to #593

### CODE REVIEW

**Comment by sijad** [comment]
should I work on work on `existsSync`? it will make testing way easier I guess

**Comment by ry** [comment]
Hi @sijad Thanks! You're binding isn't passing on Windows. Could you investigate it?

Regarding `existsSync` - that function is a special case of `statSync`. I'd rather you left that one for me to do....

**Comment by sijad** [comment]
I'm not sure how it's possible to make test works in windows without `makeTempDir`,
currently it checks that error must be equal to `deno.PermissionDenied` for unix which is not current either.

`makeTempDir` and `statSync` are required for proper testing I guess.

**Comment by ry** [comment]
@sijad Ya - we're still trying to bootstrap - if you write them and just leave it commented out, I'm okay with that.

**Comment by sijad** [comment]
tests are commented for now.

**Comment by sijad** [comment]
now it has better tests thanks to `makeTempDirSync` and `statSync`.
I think this is ready for review.

File: js/os_test.ts [hunk 1]
```diff
@@ -183,3 +183,24 @@ test(function makeTempDirSyncPerm() {
...
+  const path = deno.makeTempDirSync() + "/dir/subdir";
+  deno.mkdirSync(path);
+  const pathInfo = deno.statSync(path);
+  assert(pathInfo.isDirectory());
+});
+
+testPerm({ write: false }, function mkdDirSyncPerm() {
+  let err;
+  try {
+    const path = deno.makeTempDirSync() + "/dir/subdir";
```
Comment (ry) [inline js/os_test.ts]: Remove this line - currently the test is only testing the permissions of makeTempDirSync

**Pull request merged**